docs/proposals/observability.md

```diff
@@ -0,0 +1,311 @@
+# Enhancement Proposal-1778: Observability Policy
+
+- Issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1778
+- Status: Implementable
+
+## Summary
+
+This Enhancement Proposal introduces the `ObservabilityPolicy` API, which allows Application Developers to define settings related to tracing, metrics, or logging at the HTTPRoute level.
+
+## Goals
+
+- Define the Observability policy.
+- Define an API for the Observability policy.
+
+## Non-Goals
+
+- Provide implementation details for implementing the Observability policy.
+
+## Introduction
+
+### Observability Policy
+
+The Observability Policy contains settings to configure NGINX to expose information through tracing, metrics, and/or logging. This is a Direct Policy that is attached to an HTTPRoute or HTTPRoute Rule by an Application Developer. It works in conjunction with a [Gateway Settings](gateway-settings.md) configuration that contains higher level settings to enable Observability at this lower level. The [Gateway Settings](gateway-settings.md) configuration is managed by a Cluster Operator.
+
+Since this policy is attached to an HTTPRoute or HTTPRoute rule, the Observability settings should just apply to the relevant `location` contexts of the NGINX config for that route or rule.
+
+To begin, the Observability Policy will include the following NGINX directives (focusing on OpenTelemetry tracing):
+
+- [`otel_trace`](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace): enable tracing and set sampler rate
+- [`otel_trace_context`](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace_context): export, inject, propagate, ignore.
+- [`otel_span_name`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_name)
+- [`otel_span_attr`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_attr)
+
+Tracing will be disabled by default. The Application Developer will be able to use this Policy to enable and configure tracing for their routes. This Policy will only be applied if the OTel endpoint has been set by the Cluster Operator on the [Gateway Settings](gateway-settings.md).
+
+Ratio and parent-based tracing should be supported as shown in the [nginx-otel examples](https://github.com/nginxinc/nginx-otel?tab=readme-ov-file#examples).
+
+In the future, this config will be extended to support other functionality, such as those defined in the [NGINX Extensions Proposal](nginx-extensions.md#observability).
+
+## API, Customer Driven Interfaces, and User Experience
+
+The `ObservabilityPolicy` API is a CRD that is a part of the `gateway.nginx.org` Group. It is a namespaced resource that will reference an HTTPRoute or HTTPRoute Rule as its target.
```

Comment:
Just to double check, will it also work ok with GRPCRoute?

Reply:
I think we first need to verify that the module itself works with gRPC, which I don't have an answer to right now.